Tests/SentryTests/SentryCurrentDateTests.swift

```diff
@@ -0,0 +1,27 @@
+import XCTest
+@testable import Sentry
+
+class SentryCurrentDateTests: XCTestCase {
+    
+    override func tearDown() {
+        CurrentDate.setCurrentDateProvider(DefaultCurrentDateProvider())
```

Comment:
Can we get around just having this provider instance based?
This static isn't great for testing either since once we parallelize test execution we'll have flakey tests.

Reply:
I understand it's simpler when the dependency is hidden (not passed explicitly) and this can be achieved by relying on a static method. It doesn't solve the issue that now anywhere in the process that state can be changed and thus resulting in flakey tests or otherwise prod code.

It is "better than before" since now at least it's testable, but I'm just pointing out that it's a potential liability going forward. 
I'm also not against merging this, I just would like to communicate this and have us discuss the trade-offs and possible other approaches before we go forward with this.

Here's an example in dealing with the same problem in the .NET SDK:

https://github.com/getsentry/sentry-dotnet/blob/1d62d6bec34acc0ec7b210c63833f1c0b5cf69a8/src/Sentry/Internal/Http/SentryHeaders.cs#L26

The "clock" is passed as a dependency (with an optional parameter which we can't use here, but we can overload).
For testing, we can use the overload passing the "mock" clock provider. On prod code, we use the second overload that doesn't take clock and relies on the singleton that implements that interface and points to the static system clock:

https://github.com/getsentry/sentry-dotnet/blob/1d62d6bec34acc0ec7b210c63833f1c0b5cf69a8/src/Sentry/Internal/Http/SentryHeaders.cs#L28

Now the advantage here is that you can test this class in total isolation. Without having to mutate global state (the static DateProvider). And it stays simple for prod code that doesn't pass a clock and relies on the static one.